### ISSUE TITLE
obs_dim

### ISSUE THREAD

**Issue by Kai-Sheng-caesar** [intuitive-robots/MoDE_Diffusion_Policy#9]
In the MoDE_Diffusion_Policy/conf/config_libero.yaml, obs_dim: 512, but in the MoDE_Diffusion_Policy/conf/model/mode_agent.yaml, obs_dim: 2048.
Will these two settings cause conflicts during training? Should I unify these two settings and what value do I need to set? 
Looking forward to your reply！

**Comment by mbreuss** [comment]
Hi sorry for the confusion. The obs dim in the main config is a bit of legacy code. In the mode agent itself we define the obs dim based on the used ResNet. The 2048 is for using ResNet-50, the old 512 for the smaller ResNet-18. I would recommend to set 2048 for default ResNet-50. This obs_dim is just for the projection layer from the ResNet output token to the model latent dimension.

**Comment by Kai-Sheng-caesar** [comment]
Thanks for your reply!

**Comment by Kai-Sheng-caesar** [comment]
And if I want to train a new model with different num_experts and top_k, I just need to modify these two values in the yaml for training or there are other places I need to modify?

**Comment by mbreuss** [comment]
Check the load balancing parameter and use value non zero. Otherwise yes then you are good to go:) I recommend to use 512 latent dim for testing. Its smaller but still good enough.

**Comment by Kai-Sheng-caesar** [comment]
Okay, thanks!

**Comment by Kai-Sheng-caesar** [comment]
Hello, I find that the success rate(10 tasks x 20 rollouts) differs a lot when I set different seeds and train from scratch on LIBERO-10. Do you have the similar experience and how to deal with it?

**Comment by mbreuss** [comment]
Hi, for training from scratch its better do adjust some hyper parameters and the training time. 
Use 8 instead of 12 layers and set https://github.com/intuitive-robots/MoDE_Diffusion_Policy/blob/main/conf/model/mode_agent.yaml#L5 this to 0.1 again. Next train for a bit longer and eval every 5 epochs. Then the results should stabilize better around 0.92 something.

**Comment by Kai-Sheng-caesar** [comment]
Thanks!
It states that for fine-tuning, a value of 0.0 should be used, while for training from scratch, a value of 0.01 should be employed in https://github.com/intuitive-robots/MoDE_Diffusion_Policy/blob/main/conf/model/mode_agent.yaml#L5 and I have set it to 0.01 in my recent experiments. Therefore, I would like to confirm that I should set entropy_gamma to 0.1, as you indicated.
And I plan to set batch_size to 128, max_epoch to 40 now. Should I set a bigger max_epoch or 40 is enough?

**Comment by mbreuss** [comment]
sorry 0.01 is fine. Try to train up to 60 and evaluate from 30 upwards. Then you can see where the model achieves best performance.

**Comment by Kai-Sheng-caesar** [comment]
Thanks!

**Comment by Kai-Sheng-caesar** [comment]
Sorry, I'm just curious about why setting the n_layer to 8 can solve the fluctuation of the success rate. Could you please briefly explain the reason to me? Thanks!

**Comment by mbreuss** [comment]
12 layers with 4 experts is very big for small datasets like libero. You only need it for pretraining where you have a lot of data available to make use of the weights. Otherwise it's a bit too much capacity to fit small datasets like LIBERO and this often causes higher variance within training. In pretraining enough capacity is crucial to not underfit on the diverse data. However LIBERO is not diverse and an easier benchmark so this is not required.

**Comment by Kai-Sheng-caesar** [comment]
<img width="493" alt="Image" src="https://github.com/user-attachments/assets/b34d056b-4b95-44e1-8020-70b2a8ce022b" />

<img width="469" alt="Image" src="https://github.com/user-attachments/assets/1eccd697-24f8-4a4e-8fdc-a48acc6784a6" />

Thank you very much!
Should I set the "resnet_type" to"18" to choose FiLM-Resnet18 instead of FiLM-Resnet50 as the image encoder when training on LIBERO-10? And I find that when setting the "resnet_type" to"18", ResNetEncoderWithFiLM is imported actually and has no pretrained weights?

<img width="510" alt="Image" src="https://github.com/user-attachments/assets/8b431175-5a7b-483f-857b-e3d745f7dc3f" />

And if FiLM-Resnet50 is suggested, I set "pretrained" to be "False" now. Does it matter? I'd like to load the weights of FiLM-Resnet50 locally, where can I download the weights?

**Comment by mbreuss** [comment]
The pretrained weights will be loaded directly from torch, so you don't need to worry about it. I think both work fine. 
Maybe just use ResNet-34 then you have the middle one :D

**Comment by Kai-Sheng-caesar** [comment]
OK, thanks!

**Comment by Kai-Sheng-caesar** [comment]
Thank you for your great work! I have a question to ask. The CALVIN dataset is too large, and the size of the LIBERO dataset feels more manageable. Are there other suitable datasets similar to LIBERO that can be used for training and testing? Looking forward to your reply!

**Comment by mbreuss** [comment]
CALVIN D only has 140 GB roughly. Maybe that could be an easy way? Otherwise you could also filter out all parts of CALVIN that do not contain language which is more than 90% of files. So by manually doing that you can reduce the size of CALVIN.

**Comment by Kai-Sheng-caesar** [comment]
Thanks!

**Comment by Kai-Sheng-caesar** [comment]
Hello, I did not feel that the number of experts affected the final performance in my experiment on LIBERO and CALVIN D. I would like to ask if in the experiment of increasing the number of expert networks, you compare by selecting 2 out of 2, 2 out of 4, 2 out of 6 and 2 out of 8? Does the number of parameters in a single expert network remain unchanged? Is the amount of parameters used during the final inference the same? Looking forward to your reply!

**Comment by mbreuss** [comment]
Yes, I experienced the same 4 worked best in the scaling setting of abc, but changing to 6 or 8 did not help much. LIBERO is a bit too easy to gain anything from it and so changing the numbers does not really help. In order to benefit from more experts one need to change the routing to be not only conditioned on noise. if you swap back to token+noise routing its more relevant, since you can use experts better given the tokens. I also tried once to have experts with different number of experts but this was towards the end and I did not fully explore it. Same with expert that just uses a forward pass without weights. I think there is probably more potential to explore but I would recommend to use some prestrainign for that. The other sims are too easy for the model to overfit already.

**Comment by Kai-Sheng-caesar** [comment]
OK, thanks!

**Comment by Kai-Sheng-caesar** [comment]
<img width="200" alt="Image" src="https://github.com/user-attachments/assets/8deeb843-a831-4b09-b059-d4a11af2fc34" />

<img width="200" alt="Image" src="https://github.com/user-attachments/assets/8a782c31-6a90-45e3-b4be-a1f194d420e9" />

Hello, thanks for your great work!
In my experiments, I noticed that the results(Avg. Len.) of using different seeds to train from scratch on the CALVIN D dataset will not fluctuate beyond 0.1.  However, it really fluctuated greatly on the CALVIN ABC dataset, and the difference of Avg. Len. can be more than 0.5. Such fluctuations will not occur based on Table1 in your paper. But in Figure 8, there are also significant fluctuations as I mentioned. Seed142 and 242 have similar results.  And seed42 and 342 have similar results. But the results of seed42 and 342 will be significantly lower than seed142 and 242.
I would like to ask if you have ever encountered such a situation? What is the reason for this, and how should it be resolved? Looking forward to your reply!

**Comment by mbreuss** [comment]
Hi,

0.5 seems very big Did you try to lowering the LR rate and testing it running the policy several times during training to observe how stable the success rate across training time is? I just run 3 seeds for the final results on ABC that were fine as you can find in the HF logs. Maybe check them out to see how my performance flucates and if thats very differnt to your experiences. Sorry for that. The figure 8 was done with older models that were a bit less stable earlier in the project.

**Comment by Kai-Sheng-caesar** [comment]
Thanks！
